modulus/sym/eq/pdes/navier_stokes.py

```diff
@@ -339,6 +339,90 @@ def __init__(self, T, dim=3, time=True):
             normal_x * T.diff(x) + normal_y * T.diff(y) + normal_z * T.diff(z)
         )
 
+class NavierStokesIncompressible(PDE):
+    '''Cystom implementation of incompressible NS eqyastyans'''
+
+    name = "navier stokes incompressible"
+
+    def __init__(self, nu, rho=1, dim=3, time=False):
+
+        '''Parameters
+        ==========
+        nu : float, Sympy Symbol/Expr, str
+            The kinematic viscosity. If `nu` is a str then it is
+            converted to Sympy Function of form `nu(x,y,z,t)`.
+            If `nu` is a Sympy Symbol or Expression then this
+            is substituted into the equation. This allows for
+            variable viscosity.
+        rho : float, Sympy Symbol/Expr, str
+            The density of the fluid. If `rho` is a str then it is
+            converted to Sympy Function of form 'rho(x,y,z,t)'.
+            If 'rho' is a Sympy Symbol or Expression then this
+            is substituted into the equation to allow for
+            compressible Navier Stokes. Default is 1.
+        dim : int
+            Dimension of the Navier Stokes (2 or 3). Default is 3.
+        time : bool
+            If time-dependent equations or not. Default is False.'''
+
+        self.dim = dim
+        self.time = time
+
+        # coordinates
+        x, y, z = Symbol("x"), Symbol("y"), Symbol("z")
+
+        # time
+        t = Symbol("t")
+
+        input_variables = {"x": x, "y": y, "z": z, "t": t}
+        if self.dim == 2:
+            input_variables.pop("z")
+        if not self.time:
+            input_variables.pop("t")
+
+
+         # kinematic viscosity
+        if isinstance(nu, str):
+            nu = Function(nu)(*input_variables)
+        elif isinstance(nu, (float, int)):
+            nu = Number(nu)
+
+        # density
+        if isinstance(rho, str):
+            rho = Function(rho)(*input_variables)
```

Comment:
If this incompressible, then surely rho must be a constant? Therefore raising an exception would make more sense

Reply:
You probably correct, I will fix it